### PR TITLE
Add docs for embedded registry mirror

### PR DESCRIPTION
**Pull request by brandond**
Add docs for spegel embedded registry mirror, and tidy other docs sections that touch on registries or airgap.

### CODE REVIEW

File: docs/installation/airgap.md [hunk 1]
```diff
@@ -5,35 +5,54 @@ weight: 60
...
+  sudo mkdir -p /var/lib/rancher/k3s/agent/images/
+  sudo curl -L -O /var/lib/rancher/k3s/agent/images/k3s-airgap-images-amd64.tar.zst https://github.com/k3s-io/k3s/releases/download/v1.29.1-rc2%2Bk3s1/k3s-airgap-images-amd64.tar.zst
+  ```
+3. Proceed to the [Install K3s](#install-k3s) section below.
 
-```bash
-sudo mkdir -p /var/lib/rancher/k3s/agent/images/
-sudo cp ./k3s-airgap-images-$ARCH.tar /var/lib/rancher/k3s/agent/images/
-```
+### Embedded Registry Mirror
```
Comment (rancher-max) [inline docs/installation/airgap.md:44]: Can we list this as "experimental" in the docs for now as well?

Comment (dereknola) [reply docs/installation/airgap.md:44]: I second this

File: docs/installation/registry-mirror.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+Be aware that the embedded registry may make available images that a node may not otherwise have access to.
+For example, if some of your images are pulled from a registry, project, or repository that requires authentication via Kubernetes Image Pull Secrets, or credentials in `registries.yaml`,
+the distributed registry will allow other nodes to share those images without providing any credentials to the upstream registry.
+
+Users with access to push images into the containerd image store on one node may be able to use this to 'poison' the image for other cluster nodes,
+as other nodes will trust the tag advertised by the node, and use it without checking with the upstream registry.
+If image integrity is important, you should use image digests instead of tags, as the digest cannot be poisoned in this manner.
+
+The distributed registry is built on peer-to-peer networking, and assumes an equal level of privilege and trust between all cluster members.
+If this does not match your cluster's security posture, you should not enable the embedded distributed registry.
```
Comment (dereknola) [inline docs/installation/registry-mirror.md:86]: I would consider wrapping this in a :::warning admonition, just to get the TLDR point across.

**Review by dereknola — CHANGES_REQUESTED** [review]
Need to add this flag to the https://docs.k3s.io/cli/server#critical-configuration-values

**Review by dereknola — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Review by rancher-max — APPROVED** [review]

**Pull request merged**